internal/zinc-core/src/main/scala/sbt/internal/inc/IncrementalNameHashing.scala

```diff
@@ -32,8 +32,10 @@ private final class IncrementalNameHashing(log: sbt.util.Logger, options: IncOpt
       val aNameHashes = a.nameHashes
       val bNameHashes = b.nameHashes
       val modifiedNames = ModifiedNames.compareTwoNameHashes(aNameHashes, bNameHashes)
-      val apiChange = NamesChange(className, modifiedNames)
-      Some(apiChange)
+      if (modifiedNames.regularNames.nonEmpty || modifiedNames.implicitNames.nonEmpty) {
```

Comment:
Good catch - I recall removing that assertion probably it was lost in rebasing process.

I will remove that new check and if I face another problem with it I will create new PR.